odoo/addons/base/models/ir_sequence.py

```diff
@@ -196,9 +196,9 @@ def _interpolate(s, d):
         def _interpolation_dict():
             now = range_date = effective_date = datetime.now(pytz.timezone(self._context.get('tz') or 'UTC'))
             if self._context.get('ir_sequence_date'):
-                effective_date = datetime.strptime(self._context.get('ir_sequence_date'), '%Y-%m-%d')
+                effective_date = self._context.get('ir_sequence_date')
```

Comment:
Use Datetime.from_string() because we don't know for sure where the context value comes from.